runtime/matchers.go

```diff
@@ -0,0 +1,39 @@
+package runtime
+
+import "strings"
+
+// EqualMatcher performs a case-sensitive equality match for request metadata keys
+func EqualMatcher(s string) ServeMuxOption {
```

Comment:
@tmc , I am not sure what you mean by `WithHeaderMatcher` option? Mind elaborating a bit?